test/bats/test.bats

```diff
@@ -300,7 +301,46 @@ __namespace_exclusion_test() {
   run kubectl apply -f test/externaldata/dummy-provider/policy/examples/valid.yaml
   assert_success
 
+  # mutation test
+  run kubectl apply -f test/externaldata/dummy-provider/mutation/valid.yaml
+  assert_success
+
+  run kubectl run nginx --image=nginx --dry-run=server --output json
+  assert_success
+  assert_match "kubernetes-admin_valid" "$(jq -r '.metadata.annotations["external-data-username"]' <<< ${output})"
+  assert_match "nginx_valid" "$(jq -r '.spec.containers[0].image' <<< ${output})"
+  assert_match "busybox_valid" "$(jq -r '.spec.containers[1].image' <<< ${output})"
+
+  run kubectl apply -f test/externaldata/dummy-provider/mutation/invalid_assignmetadata.yaml
+  assert_match 'only username data source is supported' "${output}"
+  assert_match 'invalid location' "${output}"
+  assert_failure
+
+  run kubectl apply -f test/externaldata/dummy-provider/mutation/invalid_assign.yaml
+  assert_match '`default` must not be empty when `failurePolicy` is set to `UseDefault`' "${output}"
+  assert_match 'cannot assign external data response to a list' "${output}"
+  assert_failure
+
+  # simulate key error
```

Comment:
Are these error simulation tests testing mutation?

Reply:
Yes, these are mutations based on `test/externaldata/dummy-provider/mutation/valid.yaml`. I intentionally put erroneous keys (any keys with `error_` prefix and _systemError suffix are considered erroneous in https://github.com/open-policy-agent/gatekeeper/blob/master/test/externaldata/dummy-provider/provider.go#L64-L68) to the dummy provider so it can return an error.